src/contracts/HicNFT/util.ts

```diff
@@ -306,6 +326,39 @@ export async function getBalance(tezosUrl: string, mapId: number, address: strin
     return balance;
 }
 
+/**
+ *
+ *
+ */
+export async function getTokenInfo(node: Node, mapId: number = 515): Promise<{ holders: number; totalBalance: number }> {
+    const { conseilUrl, apiKey, network } = node;
+
+    let holdersQuery = ConseilQueryBuilder.blankQuery();
+    holdersQuery = ConseilQueryBuilder.addFields(holdersQuery, 'value');
+    holdersQuery = ConseilQueryBuilder.addPredicate(holdersQuery, 'big_map_id', ConseilOperator.EQ, [mapId]);
+    holdersQuery = ConseilQueryBuilder.setLimit(holdersQuery, 20_000);
+
+    const holdersResult = await TezosConseilClient.getTezosEntityData({ url: conseilUrl, apiKey, network }, network, 'big_map_contents', holdersQuery);
+
+    let holders = 0;
+    let totalBalance = new BigNumber(0);
+    holdersResult.forEach((r) => {
+        try {
+            const balance = new BigNumber(r.value);
+            if (balance.isGreaterThan(0)) {
+                totalBalance = totalBalance.plus(balance);
+            }
+            holders++;
```

Comment:
question, `holders++` is needed, cannot be holdersResult.length?

Reply:
looking for holders with >0 balances only.